Modules/_collectionsmodule.c

```diff
@@ -512,7 +512,7 @@ deque_inplace_concat(dequeobject *deque, PyObject *other)
 }
 
 static PyObject *
-deque_copy(PyObject *deque)
+deque_copy(PyObject *deque, PyObject *Py_UNUSED(ignored))
```

Comment:
Explicit cast to `PyCFunction` at line 1590 is not needed.

The same with other functions which exactly match the signature of `PyCFunction`.

Reply:
I've kept the casts in place because we will eventually need it.  To fully fix the warnings, the `ml_meth` type should be like so:

`PyObject *(*) (PyObject *, PyObject *, ...)`

so that it matches with all of the allowed method types provided that there is an explicit cast in place.  Right now I am not sure if I should change the `PyCFunction` type to that or to make a new `PyCFunctionAny` just for `ml_meth`.  Since we're on this topic, what do you think is the preferred change?  I suppose both change API, but not in a completely incompatible way.

Reply:
The casts are not needed if the signature matches `PyCFunction`. It is still needed if the function has type `PyCFunctionWithKeywords`, `PyNoArgsFunction`, etc. Removing unneeded casts will clean up the code and can prevent introducing new bugs when the signature be changed unintentionally.